.github/workflows/scala-new.yml

```diff
@@ -0,0 +1,66 @@
+name: Engine CI V2
+
+on:
+  push:
+    branches: ["*"]
+  pull_request:
+    branches: ["*"]
+  workflow_dispatch:
+
+jobs:
+  test_and_publish_win:
```

Comment:
I think the name of the job is slightly inaccurate, as it has a `win` postfix while working on both Linux and Windows. It's not blocking and can be fixed in later PRs.

Other changes look nice to me.